src/hooks/useExternalSubmitListener.ts

```diff
@@ -47,12 +47,30 @@ interface UseExternalSubmitListenerProps {
 export function useExternalSubmitListener({ formRef, patientUuid, formUuid }: UseExternalSubmitListenerProps) {
   useEffect(() => {
     const handleSubmit = (event: Event) => {
+      if (!formRef?.current) {
+        throw new Error(
+          "Form reference is missing or not attached. Ensure the 'formRef' is correctly passed and the form element is mounted.",
+        );
```

Comment:
Since the user doesn't have much control over this, I would simply log an error as opposed to throwing.  

```ts
console.error("Something went wrong: form reference is missing or not attached. Ensure 'formRef' is correctly passed and the form element is mounted.");
```

Reply:
Ideally, though, the user should get some notification that the task failed. I think we may have a notification system for something like that?

Reply:
We have a utility for reporting errors [here](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/443696ed43de6ab2297c9772e4ec4573ddc3a9b1/src/utils/error-utils.ts#L3).